spark/src/main/scala/org/apache/spark/sql/delta/skipping/clustering/ClusteringColumn.scala

```diff
@@ -40,8 +39,12 @@ object ClusteringColumn {
     val physicalNameParts = logicalNameParts.foldLeft[(DataType, Seq[String])]((schema, Nil)) {
       (partial, namePart) =>
         val (currStructType, currPhysicalNameSeq) = partial
-        val field =
-          currStructType.asInstanceOf[StructType].find(field => resolver(field.name, namePart)).get
+        val field = currStructType.asInstanceOf[StructType].find(
+          field => resolver(field.name, namePart)) match {
+          case Some(f) => f
+          case None =>
+            throw DeltaErrors.columnNotInSchemaException(logicalName, schema)
```

Comment:
Can we have a test to validate this exception?

Reply:
Added